controllers/mirroring/mirroring_controller.go

```
@@ -471,38 +472,28 @@ func (r *MirroringReconciler) reconcileRadosNamespaceMirroring(
 	*/
 	errorOccurred := false
 
-	peerClientIDs := []string{}
 	storageConsumerByName := map[string]*ocsv1alpha1.StorageConsumer{}
-	for localClientID, peerClientID := range clientMappingConfig.Data {
-		// Check if the storageConsumer with the ClientID exists, this is a fancy get operation as
-		// there will be only one consumer with the clientID
-		storageConsumers := &ocsv1alpha1.StorageConsumerList{}
-		if err := r.list(
-			storageConsumers,
-			client.MatchingFields{clientIDIndexName: localClientID},
-			client.Limit(2),
-		); err != nil {
-			r.log.Error(err, "failed to list the StorageConsumer")
-			errorOccurred = true
-			continue
-		}
-		if len(storageConsumers.Items) != 1 {
-			r.log.Error(
-				fmt.Errorf("invalid number of StorageConumers found"),
-				fmt.Sprintf("expected 1 StorageConsumer but got %v", len(storageConsumers.Items)),
-			)
-			errorOccurred = true
-			continue
+	storageConsumerByRemoteClientID := map[string]*ocsv1alpha1.StorageConsumer{}
+
+	storageConsumerList := &ocsv1alpha1.StorageConsumerList{}
+	if err := r.List(r.ctx, storageConsumerList); err != nil {
+		r.log.Error(err, "failed to list storage consumers")
+		return true
```

We need to return the error here, not true